pf9/cluster/commands.py

```diff
@@ -406,19 +432,27 @@ def attach_node(ctx, **kwargs):
     """
     logger.info(msg=click.get_current_context().info_name)
 
-    # Load active config
-    Get(ctx).active_config()
-    # Validate Token
-    Get(ctx).get_token_project_user_id()
-
     segment_session = SegmentSession()
     segment_event_properties = dict(arg_cluster_name=ctx.params['cluster_name'], arg_master_ips=ctx.params['master_ip'],
-                                    arg_worker_ips=ctx.params['worker_ip'], arg_floating_ip=ctx.params['floating_ip'],
-                                    keystone_user_id=ctx.params['user_id'],
-                                    du_account_url=ctx.params['du_url'])
+                                    arg_worker_ips=ctx.params['worker_ip'], arg_floating_ip=ctx.params['floating_ip'])
     SegmentSessionWrapper(ctx).load_segment_session(segment_session, segment_event_properties, "Attach Node")
-    SegmentSessionWrapper(ctx).send_track("Load Active config")
+
+    try:
+        # Load active config
+        Get(ctx).active_config()
+        # Validate Token
+        Get(ctx).get_token_project_user_id()
+    except Exception as e:
+        error_msg = "Encountered an error while attaching nodes to a Kubernetes" \
+                    " cluster {}. {}".format(ctx.params['cluster_name'], e.msg)
+        click.secho(error_msg, fg="red")
+        SegmentSessionWrapper(ctx).send_track_error('Load Active config', e)
+        sys.exit(1)
+
+    SegmentSessionWrapper(ctx).reload_segment_session_with_auth()
+    SegmentSessionWrapper(ctx).send_track('Load Active config')
     segment_session.send_identify(ctx.params['du_username'], ctx.params['user_id'])
+    segment_session.send_group(ctx.params['user_id'], ctx.params['du_url'])
 
     if not ctx.params['master_ip'] and not ctx.params['worker_ip']:
         msg = "No nodes were specified to be attached to the cluster {}."
```

Comment:
No need to track this error?

Reply:
I am not sure what error are you talking about. Error from send_group function will be logged inside and not a disruptive error.